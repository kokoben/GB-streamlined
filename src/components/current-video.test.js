import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CurrentVideo from './current-video';

// Retrun a fixed timestasmp when moment().format() is aclled
jest.mock('moment', () => () => ({format: () => '2018-01-30'}));


it('renders without crashing', () => {
  const mockHomeVideo = {
    name: "name",
    user: "user",
    pub_date: "pub_date",
    deck: "deck",
    embed_player: "embed_player",
  }
  const mockSetLatestHomeVideo = jest.fn();

  const wrapper = shallow(<CurrentVideo
    homeVideo={mockHomeVideo}
    name={mockHomeVideo.name}
    user={mockHomeVideo.user}
    pub_date={mockHomeVideo.pub_date}
    deck={mockHomeVideo.deck}
    embed_player={mockHomeVideo.embed_player}
    setLatestHomeVideo={mockSetLatestHomeVideo}
  />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
