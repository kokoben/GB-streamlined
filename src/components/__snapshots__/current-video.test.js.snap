// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  style={
    Object {
      "background": "#000",
      "margin": "0 auto",
      "width": "100%",
    }
  }
>
  <Row
    gutter={0}
    justify="center"
    type="flex"
  >
    <Col
      lg={8}
      xs={24}
    >
      <div
        style={
          Object {
            "color": "#fff",
            "padding": "16px",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          name
        </h2>
        <p>
          <i>
             Posted by 
            user
             | 
            2018-01-30
          </i>
        </p>
        <p>
          deck
        </p>
      </div>
    </Col>
    <Col
      lg={16}
      xs={24}
    >
      <div
        className="intrinsic-container intrinsic-container-16x9"
      >
        <iframe
          allowFullScreen={true}
          frameBorder={0}
          src="embed_player"
          title="name"
        />
      </div>
    </Col>
  </Row>
</div>
`;
